Library/Homebrew/dev-cmd/determine-test-runners.rb

```diff
@@ -0,0 +1,259 @@
+# typed: strict
```

Comment:
@carlocab IMO `typed: strict` ends up making you write stuff like `T.let(formula, Formula)` everywhere when Sorbet can already figure this out implicitly. Personally I'm pretty 👎🏻 on using it anywhere as a result. Interested in thoughts from @dduugg and @reitermarkus though.

Reply:
`T.let` is only needed for ivars in `strict` mode. It can be helpful for the reader in a lot of cases. If it gets to the point that if feels distracting, it could be a code smell that too much state is being maintained. An alternative could be to group related state into something like a [`T::Struct`](https://sorbet.org/docs/tstruct) (not in this specific case though, I would say).

Reply:
> It can be helpful for the reader in a lot of cases. If it gets to the point that if feels distracting, it could be a code smell that too much state is being maintained.

I disagree, fairly strongly. `typed: strict` or `typed: strong` for me get to the point where we're moving from "writing Ruby" to "writing another language that looks a lot like Ruby". I don't think being explicit about types in this way when most of the cases are bleedingly obvious is helpful.

Reply:
I found `typed: strict` helpful after going through multiple rounds of refactoring here. If the price I have to pay is that what I've written down isn't Ruby by another language that looks a lot like Ruby, then that seems like a reasonable tradeoff.

Reply:
@carlocab I think `typed: strict` can be useful during development but I think the regular recommendations for `T.let` when it can be inferred trivially by both the reader and Sorbet to be unnecessarily verbose. I'm open to being overruled if most others disagree, though!